docker/docker_image.go

```diff
@@ -40,20 +43,14 @@ func (i *dockerImage) RawManifest(version string) ([]byte, error) {
 }
 
 func (i *dockerImage) Manifest() (types.ImageManifest, error) {
-	// TODO(runcom): unused version param for now, default to docker v2-1
-	m, err := i.getSchema1Manifest()
-	if err != nil {
+	if err := i.retrieveRawManifest(); err != nil {
```

Comment:
BTW (independent of the rest):

If all `retrieveRawManifest` does is store the values of `i.src.GetManifest`, wouldn’t it be simpler to call it directly here, and remove the `retrieveRawManifest` function entirely?

I suppose we might want caching… but if so, perhaps it should be somehow universal. Make it a contract of `ImageSource` that data is cached, or add a `CachingImageSource`?